extensions-contrib/materialized-view-selection/src/main/java/io/druid/query/materializedview/Derivative.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.materializedview;
+
+import java.util.Objects;
+import java.util.Set;
+
+public class Derivative implements Comparable<Derivative>
+{
+  private final String name;
+  private final Set<String> columns;
+  private final long avgSizeBasedGranularity;
+
+  public Derivative(String name, Set<String> columns, long size)
+  {
+    this.name = name;
+    this.columns = columns;
```

Comment:
Looks that `name` and `columns` can't be null. Please add null checks here like `this.name = Preconditions.checkNotNull(name, "name")`.